### PR TITLE
Settings levels and timer fallback epg (matrix)

### PR DESCRIPTION
**Pull request by phunkyfish**
v4.1.0
- Added: Support settings levels via the current kodi settings level
- Added: Fallback EPG Entries for Timers
- Added: Support backend channel numbers

### CODE REVIEW

File: src/enigma2/Epg.cpp [hunk 1]
```diff
@@ -521,3 +527,41 @@ bool Epg::LoadInitialEPGForGroup(const std::shared_ptr<ChannelGroup> group)
...
+{
+  int numTransferred = 0;
+
+  CLockObject lock(m_mutex);
+  for (auto& timerBasedEntry : m_timerBasedEntries)
+  {
+    if (epgChannelId == timerBasedEntry.GetChannelId())
+    {
+      EPG_TAG broadcast;
+      memset(&broadcast, 0, sizeof(EPG_TAG));
```
Comment (ksooo) [inline src/enigma2/Epg.cpp]: `EPG_TAG broadcast = {0};` is shorter and in differnece to `memset` code will also work if `EPG_TAG` should get a vtable for whatever reason. `memset` would zero the vtable, which is... well, not a good idea.

Comment (phunkyfish) [reply src/enigma2/Epg.cpp]: Cool, memset if used a lot of the existing codebase. I'll do a once through and fix them all.

**Comment by phunkyfish** [comment]
The CI build is getting many strange errors, kodiplatform, constexpr etc. Are there other branches to change before this builds. Will do a full rebuild on my side to check all is good here.

**Comment by phunkyfish** [comment]
NVM - silly mistake on my side

EDIT: not my silly mistake it's something else.

**Comment by ksooo** [comment]
Ignore CI errors for now. Not your fault.

**Comment by ksooo** [comment]
CI errors for Matrix, I mean. Leia has to build error-free.

File: src/enigma2/Epg.cpp (hunk 1 above)
Comment (phunkyfish) [reply src/enigma2/Epg.cpp]: Done

**Comment by phunkyfish** [comment]
Ok, that should be ready now.

**Comment by phunkyfish** [comment]
This is ready now.

**Review by ksooo — APPROVED** [review]
Nice.

File: pvr.vuplus/addon.xml.in [hunk 2]
```diff
@@ -175,123 +175,12 @@
...
-- Added: New setting to enable streaming over HTTPS
-- Added: New setting to enable authentication for streaming
-
-v3.12.6
-- Fixed: Windows build fix
-- Fixed: tsbuffer.ts never got deleted, fixes #115
-    </news>
-  </extension>
-</addon>
+Full changelog here: https://github.com/kodi-pvr/pvr.vuplus/blob/Matrix/pvr.vuplus/changelog.txt
```
Comment (MilhouseVH) [inline pvr.vuplus/addon.xml.in:186]: Not valid xml.

http://ix.io/1Hoc

Comment (phunkyfish) [reply pvr.vuplus/addon.xml.in:186]: Will push fix shortly.

Comment (phunkyfish) [reply pvr.vuplus/addon.xml.in:186]: #214 - Will merge now.

Comment (phunkyfish) [reply pvr.vuplus/addon.xml.in:186]: Done

**Pull request merged**